test/e2e/singlecluster/pod_test.go

```diff
@@ -522,6 +520,20 @@ var _ = ginkgo.Describe("Pod groups", func() {
 	})
 })
 
+func expectWorkloadFinalized(ctx context.Context, k8sClient client.Client, wlKey client.ObjectKey) {
+	gomega.EventuallyWithOffset(1, func(g gomega.Gomega) error {
+		var wl kueue.Workload
+		err := k8sClient.Get(ctx, wlKey, &wl)
+		if client.IgnoreNotFound(err) != nil {
+			return client.IgnoreNotFound(err)
+		}
+		if err != nil || len(wl.Finalizers) == 0 {
+			return nil
+		}
```

Comment:
By these verifications, this test succeeded even if the workload was not deleted, right?
Is this expected behavior?

Reply:
Yes, we just need to check that removed Finalizers on workload. Workload automatically removing by garbage collector after all pods deleted.

Reply:
Ah, I see. Removed finalizer indicates issued the deletion of Workload.

Reply:
It means that the deletion, when issued, will be able to complete. The delete itself is issued by the gc-controller, with the period of 20s (see [here](https://github.com/kubernetes/kubernetes/blob/a70089ab991d2d78fef14033424e8dc2dd05f13f/pkg/controller/podgc/gc_controller.go#L50)). Since removing the finalizer is the last responsibility of Kueue, then testing just that seems like a viable approach.